modules/azurite/testcontainers/azurite/__init__.py

```diff
@@ -73,7 +88,46 @@ def __init__(
         self.with_exposed_ports(blob_service_port, queue_service_port, table_service_port)
         self.with_env("AZURITE_ACCOUNTS", f"{self.account_name}:{self.account_key}")
 
-    def get_connection_string(self) -> str:
+    def get_connection_string(
+        self, connection_string_type: ConnectionStringType = ConnectionStringType.LOCALHOST
+    ) -> str:
+        """Retrieves the appropriate connection string for the Azurite container based on the specified access type.
+
+        This method acts as a dispatcher, returning a connection string optimized
+        either for access from the host machine or for inter-container communication within the same Docker network.
+
+        :param connection_string_type: The type of connection string to generate.
+                Use :attr:`ConnectionStringType.LOCALHOST` for connections
+                    from the machine running the tests (default), or
+                :attr:`ConnectionStringType.NETWORK` for connections
+                    from other containers within the same Docker network.
+        :type connection_string_type: ConnectionStringType
+        :return: The generated Azurite connection string.
+        :rtype: str
+        :raises ValueError: If an unrecognized `connection_string_type` is provided.
+        """
+        match connection_string_type:
```

Comment:
our ci currently checks for python3.9 - i would simply replace with dict lookup or elif chain